### PR TITLE
Allow docker image pull policy to be configurable

### PR DESCRIPTION
**Pull request by tnachen**
Currently docker image pull policy is hard coded in the integration, this patch allows them all to be configurable. It currently is only one flag that overrides all places images are pulled, as I don't think it's necessary to have a policy per type. Open to change if needed.

### CODE REVIEW

**Comment by ash211** [comment]
@tnachen I'm curious -- what other policy are you changing to and how are you using this flag in your workflow?

**Comment by erikerlandson** [comment]
One use case would be image development, where you just want to make sure you are getting your latest iteration on the image.  In theory you can also get this just by using `latest` tag, although I've heard there have been bugs with that, and this allows you do guaranteed pull for any tag name.

**Comment by tnachen** [comment]
I've been using "Always" as I'm making changes to my Docker image that also has data and application jar embedded in it, and so if I push a new update I want K8s to always pull the latest version of the same image and tag, otherwise it will just reuse the same image  cached,

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SparkPodInitContainerBootstrapSuite.scala [hunk 1]
```diff
@@ -27,6 +27,7 @@ import org.apache.spark.deploy.kubernetes.constants._
 class SparkPodInitContainerBootstrapSuite extends SparkFunSuite with BeforeAndAfter {
   private val OBJECT_MAPPER = new ObjectMapper()
   private val INIT_CONTAINER_IMAGE = "spark-init:latest"
+  private val DOCKER_IMAGE_PULL_POLICY = "IfNotPresent"
```
Comment (erikerlandson) [inline resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SparkPodInitContainerBootstrapSuite.scala:30]: Internally this seems like a candidate for case classes representing `IfNotPresent` or `Always`.  Either way I think we should  check for valid values on input

Comment (tnachen) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SparkPodInitContainerBootstrapSuite.scala:30]: We could if we don't foresee the values ever change. Validating here gives us the burden to maintain this list up to date with kubernetes, so if we're open to do this for this and other option moving forward I'm open to it.

Comment (foxish) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SparkPodInitContainerBootstrapSuite.scala:30]: The API does validation for us; I don't think there is a need to implement our own on top, except for nicer error messages. In any case, we can do that in a later PR if needed IMO.

**Comment by foxish** [comment]
LGTM

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SparkPodInitContainerBootstrapSuite.scala (hunk 1 above)
Comment (erikerlandson) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SparkPodInitContainerBootstrapSuite.scala:30]: If we're getting validation from the kube API that seems fine

**Pull request merged**